solver/src/main/java/org/chocosolver/solver/variables/IViewFactory.java

```diff
@@ -109,138 +108,84 @@ default BoolVar[] setBoolsView(SetVar setVar, int size, int offset) {
     //*************************************************************************************
 
     /**
-     * Creates a view based on <i>var</i>, equal to <i>var+cste</i>.
-     * @param var  an integer variable
-     * @param cste a constant (can be either negative or positive)
-     * @return an IntVar equal to <i>var+cste</i>
+     * Create an affine view based on <i>var</i>, i.e. <i>view = a * var + b</i>
+     *
+     * @param var an integer variable
+     * @param a   a coefficient
+     * @param b   a constant
+     * @return an affine view
      */
-    default IntVar intOffsetView(IntVar var, int cste) {
```

Comment:
I would prefer to keep the existing methods and replace their content by a redirection to the new intview method, in order to ensure backward compatibility (which would have avoided the rewriting of the tests).

Reply:
Personnally, I found the old names too long/verbose and it's time for a change :)
Note that the method are deprecated but still there.
An alternative could be to have shorter name, like: `offset(IntVar, int)`, `scale(IntVar, int)` and `minus(IntVar)`. (👍 / 👎 )

Reply:
Thanks! My main concern was the backward compatibility for other models. If they are still there it is fine. 

Regarding alternatives (in addition to historical names), I would suggest to harmonize with expressions, and therefore use :  
- add
- mul
- minus (minus does not already exists in the expressions but I think it would be very usefull to add it).

Reply:
We need to be sure that there is no ambiguity between (`x` is a variable ,`a` and `b` ints and `m` a model):
- `x.neg()` (which is equivalent to `minus` view), `x.mul(a)`, `x.add(b)` and
- `m.neg(x)`, `m.mul(x, a)` and `m.add(x, b)`.
Following this, we can add `m.aff(x, a, b)` (or `m.aff(a, x, b)` and probably need to detect expressions like : `x.mul(a).add(b)` to a view (which is not the easy part).

Reply:
I didn't have the neg() in mind thanks. Even if I understand your point about symetry, I would prefer m.minus(x) than m.neg(x) because it is closer to the natural language, but maybe m.neg(x) is ok too. 

Also, I sometimes wish we could have x.minus(y) to replace x.add(y.neg())... 

Then you opened two discussions

1. the naming **aff** versus **intView** => I prefer **intView**as you put in the PR, for the following reasons
- I believe that if a user uses the view instead of x.mul(a).add(b), it is because he is aware that performances my be better using views, therefore the name intView seems relevant. 
- Whereas I may say "y is cosinus x", I never say "y is an affine function of x, a, b..." so having such names in the API is not intuitive to me. 
2. the argument order => I prefer the following : a, x, b, closer to the mathematical writing and emphasing on the difference between a and b.

Reply:
I anticipated too much, I'm going to retro-pedal. 
May be we can wait for @ArthurGodet opinion before

Reply:
But, the ideal solution is to manage views through expressions, it is even more natural.
I'll think of a way to do that.
On other thing is: do we really need to keep `Settings#enableViews` which is (I suppose) never changed. 
_The discussion might concern other settings, but it is out of this PR._

Reply:
### neg or minus
The name takes its origin in the XCSP3 documentation ([the table of operators](http://xcsp.org/specifications/constraints/generic/intension/)).
So I suppose, we can keep `neg()` as the final choice.

### add `view` to API
I can live with that. I'll change this in the next commit.
The methods will be: 
- `intView(int a, IntVar var, int b)` : the default one (ie, affine)
- `absView(IntVar var)`
- `addView(IntVar var, int b)` : I would have preferred `offsetView` but it is not close to the expression DSL and it is quite _long_
- `mulView(IntVar var, int a)`: here again, I would have preferred `scaleView`, but OK
- `negView(IntVar var)`

### Parameters ordering
I choose: `int a, IntVar var, int b`.

### Settings#enableViews
I will integrate this in the current PR, unless someone is really against it.

### manage views through expressions
This should be done in another PR

Reply:
Thinking twice, `addView ` might be ambiguous: one can think it adds something to the model.
What do you think? `plusView`, `offsetView` or `addView` ?

Reply:
`Settings#enableViews` => that's very helpful for testing, I'll keep it 😀

Reply:
Maybe we did not understand each other. I was for keeping it like you did : add(), neg(), mul() without the view in the name. Indeed, we should say that they return views, and have the word view only in the intView() function.

But I agree that add() might be ambiguous, so maybe just offset() ?